plugins/plugin-docker/che-plugin-docker-machine/src/test/java/org/eclipse/che/plugin/docker/machine/MachineProviderImplTest.java

```diff
@@ -697,6 +697,56 @@ public void shouldBindCommonAndDevVolumesToContainerOnDevInstanceCreationFromRec
         assertEquals(actualBinds.length, expectedVolumes.size());
         assertEquals(new HashSet<>(asList(actualBinds)), new HashSet<>(expectedVolumes));
     }
+    
+    @Test
+    public void shouldBindVolumesFromToContainerOnDevInstanceCreationFromRecipe() throws Exception {
+    	
+    	List<String> cheServerVolumes = new ArrayList<>(asList("SomeVolumeName:/data/lib"));
+    	String expectedVolumesFrom = "che-server";
+    	
+    	provider = new MachineProviderBuilder().setVolumesFrom(expectedVolumesFrom)
+    									       .build();
+    	
+    	final boolean isDev = true;
+    	
+    	CheServiceImpl service = createService();
+    	service.setContainerName(expectedVolumesFrom);
+    	service.setVolumes(cheServerVolumes);
+    	createInstanceFromRecipe(isDev, service);
+    	
+        ArgumentCaptor<CreateContainerParams> argumentCaptor = ArgumentCaptor.forClass(CreateContainerParams.class);
+        verify(dockerConnector).createContainer(argumentCaptor.capture());
+        verify(dockerConnector).startContainer(any(StartContainerParams.class));
+        
+        final String[] actualVolumesFrom = argumentCaptor.getValue().getContainerConfig().getHostConfig().getVolumesFrom();
+        assertTrue(Arrays.asList(actualVolumesFrom).contains(expectedVolumesFrom));
+    }
+    
+    @Test
+    public void shouldBindVolumesFromToContainerOnDevInstanceCreationFromRecipeWhenServiceHasVolumesFromSet() throws Exception {
+    	
+    	List<String> cheServerVolumes = new ArrayList<>(asList("SomeVolumeName:/data/lib"));
+    	List<String> cheServerVolumesFrom = new ArrayList<>(asList("SomeContainerName"));
+    	String expectedVolumesFrom = "che-server";
+    	
+    	provider = new MachineProviderBuilder().setVolumesFrom(expectedVolumesFrom)
+    									       .build();
+    	
+    	final boolean isDev = true;
+    	
+    	CheServiceImpl service = createService();
+    	service.setContainerName(expectedVolumesFrom);
+    	service.setVolumes(cheServerVolumes);
+    	service.setVolumesFrom(cheServerVolumesFrom);
+    	createInstanceFromRecipe(isDev, service);
+    	
+        ArgumentCaptor<CreateContainerParams> argumentCaptor = ArgumentCaptor.forClass(CreateContainerParams.class);
+        verify(dockerConnector).createContainer(argumentCaptor.capture());
+        verify(dockerConnector).startContainer(any(StartContainerParams.class));
+        
+        final String[] actualVolumesFrom = argumentCaptor.getValue().getContainerConfig().getHostConfig().getVolumesFrom();
+        assertTrue(Arrays.asList(actualVolumesFrom).contains(expectedVolumesFrom));
```

Comment:
Sorry I confused you because of my mistake. I'll try to explain what I meant and where is my mistake.

MachineProviderImpl accepts in constructor some additional settings, e.g. volumes, /etc/hosts entries, env variables, etc. So these values should be added into every container (or every dev-machine container). They are system-wide.
On the other hand there is CheServiceImpl which represents particular machine configuration.
So if CheServiceImpl and MachineProviderImpl have volumes configured they should be combined in getHostConfig.
But I missed that MachineProviderImpl doesn't accept volumesFrom.

So my comment is not relevant. I'm really sorry that I confused you.